scikeras/wrappers.py

```diff
@@ -392,46 +386,45 @@ def _fit_keras_model(self, X, y, sample_weight, warm_start):
         else:
             hist = self.model_.fit(x=X, y=y, **fit_args)
 
-        if warm_start:
-            if not hasattr(self, "history_"):
-                self.history_ = defaultdict(list)
-            self.history_ = {
-                get_metric_full_name(k): self.history_[get_metric_full_name(k)]
-                + hist.history[k]
-                for k in hist.history.keys()
-            }
+        if not warm_start or not hasattr(self, "history_"):
+            self.history_ = dict()
+            for key, val in hist.history.items():
+                try:
+                    key = metric_name(key)
+                except ValueError:
```

Comment:
`metric_name` raises a `ValueError` for unknown loss functions. In that case, we want to just let them through without renaming.

The only reason renaming exists is because if you use a shorthand (`mse`) and then train you now have `mse` as the key in `self.history_`. But when you do a serialize->deserialize roundtrip, Keras uses the full name, so now your model's metric is called `mean_squeared_error`. Then when you train again via `partial_fit` you end up with both `mse` and `mean_squared_error` as keys.

I'll add a clarifying comment.

Reply:
I don't know if I'd worry about having different keys in `self.history_`. Would they be repeat values or would `"mse"` be replaced with `"mean_squared_error"`?

Reply:
They'd be split, not replaced (if by replaced you meant that `mse` gets deleted). So some of your epochs would end up in `mse` and some in `mean_squared_error`. That would be very confusing I imagine.